sources/lib/run-time/demangle.c

```diff
@@ -0,0 +1,137 @@
+#include "demangle.h"
+
+#include <limits.h>
+#include <stdbool.h>
+
+#define CONSTANT_PREFIX         'K'
+#define DYLAN_MODULE_SEP        'K'
+#define MODULE_SEP              'Y'
+#define LIBRARY_SEP             'V'
+#define ESCAPE_SEP              'Z'
+#define METHOD_MARK             'M'
+#define IEP_MARK                'I'
+
+#define ARRAY_LEN(x)            (sizeof(x)/sizeof((x)[0]))
+
+// This is adequte for IEPs, but could be improved for other things
+int dylan_demangle(char *dest, size_t destsize, char *src)
```

Comment:
How about linking to https://opendylan.org/documentation/hacker-guide/runtime/mangling.html here?